colbert/index_updater.py

```diff
@@ -0,0 +1,178 @@
+import os
+import ujson
+import torch
+import numpy as np
+import tqdm
+
+from colbert.search.index_loader import IndexLoader
+from colbert.indexing.index_saver import IndexSaver
+from colbert.indexing.collection_encoder import CollectionEncoder
+
+from colbert.utils.utils import lengths2offsets, print_message, dotdict, flatten
+from colbert.indexing.codecs.residual import ResidualCodec
+from colbert.indexing.utils import optimize_ivf
+from colbert.search.strided_tensor import StridedTensor
+from colbert.modeling.checkpoint import Checkpoint
+from colbert.utils.utils import print_message, batch
+from colbert.data import Collection
+from colbert.indexing.codecs.residual_embeddings import ResidualEmbeddings
+from colbert.indexing.utils import optimize_ivf
+
+class IndexUpdater:
+    
+    def __init__(self, config, searcher, checkpoint=None):
+        self.config = config
+        self.searcher = searcher
+        if checkpoint:
+            self.checkpoint = Checkpoint(checkpoint, config)
+            self.encoder = CollectionEncoder(config, self.checkpoint)
+        self.index_path = searcher.index
+        self._load_disk_ivf()
+        
+        
+    def remove(self, pids, persist_to_disk=False):
```

Comment:
I wonder if `persist_to_disk` should be a separate function that commits all changes made during the lifetime of the `IndexUpdater` object rather than persisting each update individually. @okhat what do you think?

Reply:
Actually discussed this offline, we will make this change

Reply:
Awesome!